ui/tests/e2e/service_mesh_performance.spec.js

```diff
@@ -1,62 +1,80 @@
+      });
+      await page.getByRole('button', { name: 'View Results', exact: true }).click();
 
-  test('Run a performance test', async ({ performancePage }) => {
-    await performancePage.runPerformanceTest(profileName);
-  });
+      expect(page.getByText('Sorry, no matching records')).toBeHidden();
+      await page.getByLabel('more').first().click();
 
-  test('View Results from a performance profile', async ({ performancePage }) => {
-    await performancePage.goToProfiles();
-    await performancePage.viewPerformanceProfileResult(profileName);
-  });
+      await expect(await page.getByText(`URL: ${url}`, { exact: true })).toBeVisible();
+    });
+
+    test(`View/Edit the configuration of a performance profile with load generator "${loadGenerator}" and service mesh "${serviceMesh}"`, async ({
+      page,
+    }) => {
+      await page.goto(`${ENV.MESHERY_SERVER_URL}/performance/profiles`);
+      await page.getByText(profileName).first().click();
+
+      await page.getByTestId('performanceProfieCard-edit').click();
+      await page.getByLabel('Service Mesh').click();
+      await page.locator(`[data-value=${serviceMesh}]`).click();
+      await page.getByRole('spinbutton', { name: 'Concurrent requests' }).fill('3');
+      await page.getByLabel(loadGenerator).check();
+
+      await expect(page.getByRole('button', { name: 'Run Test', exact: true })).toBeVisible();
+      await page.getByRole('button', { name: 'Run Test', exact: true }).click();
+
+      await expect(page.getByText('fetched the data.')).toBeVisible({ timeout: 2 * 60 * 1000 });
+    });
+
+    test(`Delete a performance profile with load generator "${loadGenerator}" and service mesh "${serviceMesh}"`, async ({
+      page,
+    }) => {
+      await page.goto(`${ENV.MESHERY_SERVER_URL}/performance/profiles`);
+      await page.getByText(profileName, { exact: true }).click();
+      await page.getByTestId('performanceProfieCard-delete').first().click();
 
-  test('View/Edit the configuration of a performance profile', async ({ performancePage }) => {
-    await performancePage.goToProfiles();
-    await performancePage.viewPerformanceProfileConfiguration(profileName);
-    await performancePage.page.locator('[aria-labelledby="meshName-label meshName"]').click();
-    await performancePage.page.locator('[data-value="istio"]').click();
-    await performancePage.fillInput(
-      performancePage.page.getByRole('spinbutton', { name: 'Concurrent requests' }),
-      '6',
-    );
-    await expect(
-      performancePage.page.getByRole('button', { name: 'Run Test', exact: true }),
-    ).toBeVisible();
-    await performancePage.page.getByRole('button', { name: 'Run Test', exact: true }).click();
-    const notification = await performancePage.page
-      .locator('text=Initiating load test . . .')
-      .first();
-    await expect(notification).toBeVisible();
+      await expect(await page.getByText('Performance Profile Deleted!').first()).toBeHidden();
```

Comment:
Should we be looking for the missing it test result by ID or test result by name instead of a message back? Asserting on the text message back seems fragile. Looking at it, I can tell you that I would like for it to be changed and I have the!m exclamation mark removed. If we were to go do that, Then the test will break.